lib/remote/filterutility.cpp

```diff
@@ -172,11 +172,35 @@ void FilterUtility::CheckPermission(const ApiUser::Ptr& user, const String& perm
 	if (!foundPermission) {
 		Log(LogWarning, "FilterUtility")
 			<< "Missing permission: " << requiredPermission;
+	}
+
+	return foundPermission;
+}
 
-		BOOST_THROW_EXCEPTION(ScriptError("Missing permission: " + requiredPermission));
+void FilterUtility::CheckPermission(const ApiUser::Ptr& user, const String& permission, Expression **permissionFilter)
+{
+	if (!MatchPermission(user, permission, permissionFilter)) {
+		BOOST_THROW_EXCEPTION(ScriptError("Missing permission: " + permission.ToLower()));
 	}
 }
 
+/**
+ * Checks whether the given API user is granted the given permission
+ *
+ * This is a wake type of the CheckPermission() method and does always return a boolean.
+ * If you want an exception to be raised when the user doesn't have the given permission,
+ * you need to use the other method, which performs a stricter permission check.
+ *
+ * @param user ApiUser pointer to the user object you want to check the permission of
+ * @param permission The actual permission you want to check the user permission against
+ *
+ * @return bool
+ */
+bool FilterUtility::HasPermission(const ApiUser::Ptr& user, const String& permission, Expression **permissionFilter)
+{
+	return MatchPermission(user, permission, permissionFilter);
```

Comment:
What's the purpose of this separate function?

Reply:
Isn't it in the docblock!

Reply:
Oh, that's what you mean! I introduced this function before I added this filter evaluation on the joined relations. However, now, we can merge them both, of course.